### PR TITLE
Separate key check from validation

### PR DESCRIPTION
**Pull request by Shelob9**
#5853

### CODE REVIEW

File: includes/api/class-edd-api.php [hunk 1]
```diff
@@ -329,7 +329,13 @@ private function validate_request() {
 				$secret = $this->get_user_secret_key( $user );
 				$public = urldecode( $wp_query->query_vars['key'] );
 
-				$this->check_keys( $secret, $public, $token );
+				$valid = $this->check_keys( $secret, $public, $token );
+				if(  $valid ){
```
Comment (sunnyratilal) [inline includes/api/class-edd-api.php]: Code formatting here just needs slightly cleaning up to read `if ( $valid ) {`

File: includes/api/class-edd-api.php [hunk 2]
```diff
@@ -2267,4 +2263,23 @@ public function log_requests() {
 		return apply_filters( 'edd_api_log_requests', true );
 	}
 
+	/**
+	 * Check API keys vs token
+	 *
+	 * @access public
+	 * @since  2.7.12
```
Comment (sunnyratilal) [inline includes/api/class-edd-api.php]: `@since` needs updating to `2.8.2`

**Comment by pippinsplugins** [comment]
@sunnyratilal could you please go ahead and set up an issue branch for this, merge it in there, then fix the formatting and since tags?

**Comment by sunnyratilal** [comment]
@Shelob9 I'm going to close out this PR - you're commit history will remain intact.

**Comment by sunnyratilal** [comment]
New PR opened at #5912

**Comment by Shelob9** [comment]
@sunnyratilal Sorry I was going to do the changes today. New PR is fine though. Thanks for getting this merged!

**Pull request closed without merging**